File: pgm/ABC_Java/ABC_225_A.java
import java.io.*;
// import java.math.*;
import java.util.*;
// import java.util.Map.*;
// import java.util.stream.*;

public class Main {
    private static int factorial (int n) {
        if (n > 0)
            return n * factorial (n - 1);
        else
            return 1;
    }
    public static void main (String[] args) throws Exception {
        BufferedReader reader =
            new BufferedReader (new InputStreamReader (System.in));
        final String[] S = reader.readLine().split ("");
        final int n = S.length;
        final int one = 1;
        Map<String, Integer> m = new HashMap<>();
        for (int i = 0; i < n; i++) {
            m.put (S[i], m.getOrDefault (S[i], 0) + 1);
        }
        int divisor = one;
        for (int i : m.values()) {
            divisor *= factorial (i);
        }
        System.out.println (factorial (n) / divisor);
        return;
    }
}

File: pgm/ABC_Kotlin/ABC_225_A.kt
// import java.io.BufferedOutputStream
// import java.io.BufferedReader
// import java.io.DataInputStream
// import java.io.FileInputStream
// import java.io.IOException
// import java.io.InputStreamReader
// import java.io.PrintWriter
// import java.lang.StringBuilder
// import java.time.LocalTime
// import java.time.format.DateTimeFormatter
// import java.util.Scanner
// import java.util.StringTokenizer

// import kotlin.math.*

const val one: Long = 1L

fun Int.factorial(): Long = (1..this).fold(1L) { acc, i -> acc * i }

fun main() {
    val S: String = readln()
    val len: Int = S.length
    val cnt: Collection<Int> = S.groupingBy { it }.eachCount().values
    val divisor: Long = cnt.fold(one) { accum, cnt -> accum * cnt.factorial() }
    println(len.factorial() / divisor)
    kotlin.system.exitProcess(0)
}